node/storage-cleaner/src/appwrite.js

```diff
@@ -0,0 +1,39 @@
+import { Client, Storage, Query } from 'node-appwrite';
+import { getExpiryDate } from './util.js';
+
+class AppwriteService {
+  constructor() {
+    const client = new Client()
+      .setEndpoint(
+        process.env.APPWRITE_ENDPOINT ?? 'https://cloud.appwrite.io/v1'
+      )
+      .setProject(process.env.APPWRITE_FUNCTION_PROJECT_ID)
+      .setKey(process.env.APPWRITE_API_KEY);
+    this.storage = new Storage(client);
+  }
+
+  /**
+   * Clean up files from the storage bucket by removing files older than a specified retention period.
+   *
+   * @param {string} bucketId - The ID of the storage bucket to clean.
+   * @returns {Promise<void>} A Promise that resolves when the bucket is cleaned.
+   */
+  async cleanBucket(bucketId) {
+    let hasNextPage = true;
+    const queries = [
+      Query.lessThan('$createdAt', getExpiryDate()),
+      Query.limit(25),
+    ];
+    do {
+      const response = await this.storage.listFiles(bucketId, queries);
+      await Promise.all(
+        response.files.map((file) =>
+          this.storage.deleteFile(bucketId, file.$id)
+        )
+      );
+      hasNextPage = response.files.length > 0;
```

Comment:
Can we move this condition inline?

e.g. `} while (response.files.length > 0);`

Reply:
Done